src/scan-iterator.ts

```diff
@@ -148,29 +148,38 @@ async function load<V>(
 ): Promise<V[]> {
   const items: V[] = [];
   type MaybeIndexName = {indexName?: string};
-  const key = (primaryKey: string, secondaryKey: string | null) =>
+
+  const toKey =
```

Comment:
I did some refactoring in here... Maybe I should have refrained from doing it in this PR

Please take a look.